sdk/compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -832,29 +835,46 @@ writeDalfFile dalfFile mod = do
                 Proto.toLazyByteString $
                     encodeSinglePackageModule lfVersion mod
 
+-- this is a lot of qualification, which to import?
+convertUnitId :: Map.Map GHC.UnitId LF.DalfPackage -> GHC.InstalledUnitId -> LF.PackageId
+convertUnitId pkgMap id =
+  --TODO: should we handle the case where lookup fails?
+  let LF.DalfPackage { dalfPackageId } = fromJust (Map.lookup (DefiniteUnitId (DefUnitId id)) pkgMap)
+  in dalfPackageId
+
+-- this is a lot of qualification, which to import?
+depsToIds :: Map.Map GHC.UnitId LF.DalfPackage -> IntMap.IntMap (Set.Set GHC.InstalledUnitId) -> LF.PackageIds
+depsToIds pkgMap unitMap = Set.map (convertUnitId pkgMap) $ mconcat $ IntMap.elems unitMap
+
 -- Generates a Daml-LF archive without adding serializability information
 -- or type checking it. This must only be used for debugging/testing.
 generateRawPackageRule :: Options -> Rules ()
 generateRawPackageRule options =
     define $ \GenerateRawPackage file -> do
         lfVersion <- getDamlLfVersion
+        PackageMap pkgMap <- use_ GeneratePackageMap file
+        deps <- depPkgDeps <$> use_ GetDependencyInformation file
+        let ids = depsToIds pkgMap deps
```

Comment:
Since this pattern shows up twice and nowhere else, I think you should move it to its own rule, e.g. `GetImports`, with these three lines as the implementation body.